### PR TITLE
Separate plugin and server, rework API (to v3)

### PR DESCRIPTION
**Pull request by kyngs**
This PR attempts to decouple the plugin from the server.
The API has been reworked to make it simpler. The loaders have been put into a separate module the developers can depend on, or provide their own.

TO-DO:
- ~~review~~
- ~~set up the loaders module publishing~~ Going to be handled after merge
- ~~properly document (see #109)~~ done
- ~~wait for 1.21~~ We're going to base 1.21 off of this

### CODE REVIEW

File: api/src/main/java/com/infernalsuite/aswm/api/AdvancedSlimePaper.java
Comment (Paul19988) [inline api/src/main/java/com/infernalsuite/aswm/api/AdvancedSlimePaper.java:1]: Could name this file "AdvancedSlimePaperAPI" just to be slightly more clear

File: core/src/main/java/com/infernalsuite/aswm/skeleton/SkeletonCloning.java [hunk 1]
```diff
@@ -19,7 +19,7 @@ public class SkeletonCloning {
     public static SkeletonSlimeWorld fullClone(String worldName, SlimeWorld world, SlimeLoader loader) {
         return new SkeletonSlimeWorld(worldName,
                 loader == null ? world.getLoader() : loader,
-                world.isReadOnly(),
+                loader == null,
```
Comment (Paul19988) [inline core/src/main/java/com/infernalsuite/aswm/skeleton/SkeletonCloning.java:22]: Are we intending on no longer providing a way to have the world be read-only?

**Review by Paul19988 — CHANGES_REQUESTED** [review]

File: core/src/main/java/com/infernalsuite/aswm/skeleton/SkeletonCloning.java (hunk 1 above)
Comment (Farmeurimmo) [reply core/src/main/java/com/infernalsuite/aswm/skeleton/SkeletonCloning.java:22]: Please don't, read-only world are perfect when you need a same spawn on every server without the need to save or edit it.

Comment (kyngs) [reply core/src/main/java/com/infernalsuite/aswm/skeleton/SkeletonCloning.java]: Good catch! This was an oversight on my side

**Review by Paul19988 — APPROVED** [review]

**Pull request merged**